src/main/kotlin/com.bugsnag.android.gradle/BugsnagManifestUuidTask.kt

```diff
@@ -7,13 +7,32 @@ import org.gradle.api.DefaultTask
 import org.gradle.api.Project
 import org.gradle.api.file.Directory
 import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.file.RegularFileProperty
 import org.gradle.api.logging.Logger
 import org.gradle.api.provider.Property
 import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.OutputFile
 import org.gradle.api.tasks.TaskAction
 import java.io.File
 import java.nio.file.Paths
-import java.util.UUID
+
+abstract class BaseBugsnagManifestUuidTask : DefaultTask() {
+    init {
+        group = BugsnagPlugin.GROUP_NAME
+        description = "Adds a unique build UUID to AndroidManifest to link proguard mappings to crash reports"
+    }
+
+    @get:Input
+    abstract val buildUuid: Property<String>
```

Comment:
My current understanding is that any property annotated with input/output is used to calculate whether a task is up-to-date or not. As the `buildUuid` input is in the base class does this mean that it will always count towards the inputs of `BugsnagManifestUuidTask/BugsnagManifestUuidTaskV2`?

Reply:
This was something Xav from the AGP team pointed out to me. We could make this up to date based on the manifest content, but we presumably want this to update every build right? It's true that the task would basically never be up to date, but I'm not sure how else to do it without adding a lot of complexity to detekt if all the other components of a build are up to date 🤔

Reply:
I think the idea would be that the plugin should be disabled on non-release builds right?

Reply:
Alternatively, we could make the builduuid generation a property of `BugsnagPluginExtension` to it could be set to something deterministic in debug builds and random on CI?

Reply:
I think it should be possible to make the `BugsnagManifestUuidV2Task` up-to-date by specifying only the two manifest files as inputs/outputs and generating the `buildUuid` within the task. However, I'm happy to address that myself in a separate changeset as this is getting a bit outside the original scope of this PR.